### ISSUE TITLE
Resource Hub

### ISSUE THREAD

**Issue by Bucknalla** [WarwickTECH/meetings#3]
- Scopes for user (owncloud?)
- Generalised user login
OAuth2

**Comment by IEavan** [comment]
Oauth2 works under the assumption that the client application is accessing a resource owned by the user and hosted by another service (in our case owncloud). The authentication interface would have to be provided by the opencloud api, which would be a little jarring for the end user, and would go against our desire to have a "Generalised user login". Additionally, since we would be running both the client application and owncloud, from oauth2's perspective, we would be requesting permission to ourselves, which is redundant. I believe it would probably be possible to implement, but it's a lot of additional complexity and will likely be slower. Considering the rough outlines of the requirements it which be simpler to have a native login system on the site to access the resource hub and then fetch the resources from with Ajax from a mySQL or other SQL database. If we go with a cloud provider like aws or azure setting this up should be easy enough.

**Comment by Bucknalla** [comment]
Makes sense! @PiR-P have you had a look into hosting? What would your preferences be for deploying something like this @IEavan? AWS?

**Comment by IEavan** [comment]
I am most familiar with aws, although any established provider should be fine

**Comment by ghost** [comment]
@Bucknalla Regarding hosting, Geoffrey should send me a complete report with comparisons by tomorrow. I'll keep you updated on that.
